### ISSUE TITLE
Local build failure OSX

### ISSUE THREAD

**Issue by zach-hensel** [greenelab/manubot-rootstock#133]
FYI a local build failure in OSX. Other hiccups encountered by this novice user going through setup -- having ruby-dev installed properly, and screwed up travis login by following the "--pro" instructions that travis gives you (because I didn't read the setup.md closely enough and tried to run a command before logging in).

OSX 10.13.6
Conda 4.5.11
Travis build is OK

```
(manubot) Zachs-MacBook-Pro:low-noise-manuscript zach$ sh build/build.sh 
Retrieving and processing reference metadata
## INFO
Manuscript content parts:
00.front-matter
01.abstract
02.introduction
03.methods
99.back-matter
## INFO
10 unique citations strings extracted from text
10 unique standard citations    
## INFO
requests-cache starting with 22 cached responses
## INFO
requests-cache finished with 22 cached responses
## INFO
Using UTC timezone.
Dating manuscript with the current datetime: 2018-08-31T13:02:51.538430+00:00
## INFO
Reading user-provided templating variables complete:
{}
## INFO
Generated manscript stats:
{
  "reference_counts": {
    "doi": 6,
    "pmid": 3,
    "url": 1,
    "total": 10
  },
  "word_count": 671
}
Exporting HTML manuscript
Traceback (most recent call last):
  File "/anaconda3/envs/manubot/bin/pandoc-eqnos", line 7, in <module>
    from pandoc_eqnos import main
  File "/anaconda3/envs/manubot/lib/python3.6/site-packages/pandoc_eqnos.py", line 44, in <module>
    import pandocxnos
  File "/anaconda3/envs/manubot/lib/python3.6/site-packages/pandocxnos/__init__.py", line 1, in <module>
    from .core import *
  File "/anaconda3/envs/manubot/lib/python3.6/site-packages/pandocxnos/core.py", line 67, in <module>
    import psutil
  File "/anaconda3/envs/manubot/lib/python3.6/site-packages/psutil/__init__.py", line 157, in <module>
    from . import _psosx as _psplatform
  File "/anaconda3/envs/manubot/lib/python3.6/site-packages/psutil/_psosx.py", line 16, in <module>
    from . import _psutil_osx as cext
ImportError: dlopen(/anaconda3/envs/manubot/lib/python3.6/site-packages/psutil/_psutil_osx.cpython-36m-darwin.so, 2): Symbol not found: _kIOMasterPortDefault
  Referenced from: /anaconda3/envs/manubot/lib/python3.6/site-packages/psutil/_psutil_osx.cpython-36m-darwin.so
  Expected in: flat namespace
 in /anaconda3/envs/manubot/lib/python3.6/site-packages/psutil/_psutil_osx.cpython-36m-darwin.so
Error running filter pandoc-eqnos:
Filter returned error status 1
```
Repo is here - https://github.com/zach-hensel/low-noise-manuscript/ - this is pretty awesome.

**Comment by agitter** [comment]
Thanks for reporting this and for trying Manubot!  As far as I know, this is a new error.  I don't run OS X, so I'll see how far I can get debugging it myself or will call upon a few other contributors to help.

**Comment by zach-hensel** [comment]
I'm very happy building w/ Travis only so will continue to use regardless; hopefully for this manuscript assuming figures and tables look OK!

**Comment by vsmalladi** [comment]
I can take a look and see if it still builds on my mac.

**Comment by agitter** [comment]
@zach-hensel can you please tell us what version of pandoc you have using `pandoc --version`?

**Comment by slochower** [comment]
I don’t have anything useful to say about this particular problem right now, but I have been regularly running Manubot on my Mac without problem. I don’t have the latest version installed though and am traveling through the weekend. I can help dig in deeper on macOS if this doesn’t get resolved.

**Comment by zach-hensel** [comment]
```
(base) Zachs-MacBook-Pro:build zach$ conda activate manubot
(manubot) Zachs-MacBook-Pro:build zach$ pandoc --version
pandoc 2.1.1
Compiled with pandoc-types 1.17.3, texmath 0.10.1, skylighting 0.6
Default user data directory: /Users/zach/.pandoc
Copyright (C) 2006-2018 John MacFarlane
Web:  http://pandoc.org
This is free software; see the source for copying conditions.
There is no warranty, not even for merchantability or fitness
for a particular purpose.
```

**Comment by agitter** [comment]
@zach-hensel that's good progress, but feel free to close the issue if you want to stay with Travis builds only.

If you want to debug further, @vsmalladi maybe able to help.  The pango issue is one we've seen before on OSX:
- https://github.com/greenelab/manubot-rootstock/issues/74#issuecomment-334349800
- https://github.com/greenelab/manubot-rootstock/issues/120#issue-309173924

Homebrew was used to install pango in those cases.  It is also available through [conda](https://anaconda.org/anaconda/pango).

**Comment by dhimmel** [comment]
Thanks @zach-hensel for trying out manubot and reporting the issues you hit.

Tagging @tomduck who creates `pandoc-xnos`, in case the psutil dependency is non-essential or can be more lazy... if such installation issues with psutil are common, it may make sense to minimize the dependency.

As for the specific reason why the psutil installation is broken, I am guessing this is some sort of conda build issue or package linking problem. These tend to be hard issues to debug and often change or get fixed with time.

Regarding the other pain points @zach-hensel mentions, I think they're helpful in guiding our longer-term architecture thinking. I've consolidated them below:

> BTW when this error occurs, I need to manually delete an images alias that's made in the manuscript root directory before I can build again.

Yes this is not ideal.

> Other hiccups encountered by this novice user going through setup -- having ruby-dev installed properly and screwed up travis login by following the "--pro" instructions that travis gives you

Yeah getting the travis ruby gem to work can be frustrating and the documentation for this utility is not great. We've [talked about dockerizing](https://github.com/greenelab/manubot-rootstock/issues/120#issuecomment-377696541) the setup environment/protocol to make it more straightforward.

Eventually the travis --pro / --org distinction will go away, and then https://github.com/greenelab/manubot-rootstock/pull/124 can be merged. This seems to be taking Travis a long time.

One possibility is to switch to GitLab for it's tighter CI integration: see https://github.com/greenelab/manubot-rootstock/issues/88.

**Comment by zach-hensel** [comment]
Update -- local build to html and pdf both work now; fix is manually installing psutil (5.6.1) and pango (1.41.0) to the manubot environment using anaconda navigator; presumably that could be added to the environment.yml instead but I didn't test.

**Comment by dhimmel** [comment]
> fix is manually installing psutil (5.6.1) and pango (1.41.0) to the manubot environment 

Cool. These packages are being installed in the current `environment.yml`:

https://github.com/manubot/rootstock/blob/b422130eb46ea5d188779910772192730d79b172/build/environment.yml#L13-L14

If you [update](https://github.com/manubot/rootstock/blob/master/SETUP.md#merging-upstream-rootstock-changes) to the latest version of rootstock and [reinstall](https://github.com/manubot/rootstock/tree/b422130eb46ea5d188779910772192730d79b172/build#environment) the conda environment, you should get this latest version.

Glad things are working!
